### ISSUE TITLE
Branch cleanup

### ISSUE THREAD

**Issue by whd** [mozilla/bigquery-etl#1650]
Per discussion with :hwoo cloning this repo takes quite a bit of time, possibly because there are many stale branches. We should probably make sure to delete branches after they've been merged and maybe delete some of the old unmerged ones.

**Comment by acmiyaguchi** [comment]
I pruned the low hanging fruit. After removing some of the stale and merged branches, there are now 50ish branches. I was under the impression that git doesn't pull down all of the branches when checking out the master branch though.

**Comment by whd** [comment]
> git doesn't pull down all of the branches when checking out the master branch though.

 AFAIK `git checkout` never fetches any remote tracking branches. We were talking about a fresh `clone` and automated airflow `pull`s of master, also see https://git-scm.com/docs/git-pull, since he was debugging https://bugzilla.mozilla.org/show_bug.cgi?id=1683750. :hwoo and I did discuss that it would probably be better to explicitly fetch/merge since we only care about one remote branch, and `git clone` supports `--single-branch`.

A `git clone` I just ran took <2s (note: didn't test my clone time before) so this is probably sufficient.
